File: Java/Easy/MaximumSubarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maximum = Integer.MIN_VALUE;
        int sum = 0;
        for(int i = 0; i < nums.length; i++){
            if(sum < 0){
                sum = nums[i];
                System.out.println("Sum: " + sum + "\nNums: " + nums[i]);
            }
            else
                sum += nums[i];
            maximum = sum > maximum ? sum: maximum;
        }
        return maximum;
    }
}

File: Sprint/MaximumSubarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        temp = nums[0]
        max_so_far = nums[0]
        for n in nums[1:]:
            temp = max(n, temp + n)
            max_so_far = max(temp, max_so_far)
        return max_so_far
            
